### PR TITLE
handle windows lineends after checkout for Esp32 builds under windows

### LINKED ISSUE #57: esp32 docker configure-esp32 failed
Hi, I'm trying to use docker esp32 on windows 10.  I followed instructions to Configure modules and features for ESP32, then got following errors with configure-esp32.

I ran:
`docker run --rm -it -v c:/Users/Daniel/nodemcu-firmware:/opt/nodemcu-firmware marcelstoer/nodemcu-build configure-esp32`

got following errors:

```
Setting IDF_PATH and re-invoking...
make[1]: Entering directory '/opt/nodemcu-firmware'
make[2]: Entering directory '/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig'
make[2]: /opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/check-lxdialog.sh: Command not found
cc -c   -DLOCALE -MMD  /opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c -o mconf.o
In file included from /opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c:23:0:
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:38:10: error: #include expects "FILENAME" or <FILENAME>
 #include CURSES_LOC
          ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:103:2: error: unknown type name 'chtype'
  chtype atr; /* Color attribute */
  ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:200:16: error: unknown type name 'WINDOW'
 int on_key_esc(WINDOW *win);
                ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:221:17: error: unknown type name 'WINDOW'
 void attr_clear(WINDOW * win, int height, int width, chtype attr);
                 ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:221:54: error: unknown type name 'chtype'
 void attr_clear(WINDOW * win, int height, int width, chtype attr);
                                                      ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:223:21: error: unknown type name 'WINDOW'
 void print_autowrap(WINDOW * win, const char *prompt, int width, int y, int x);
                     ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:224:19: error: unknown type name 'WINDOW'
 void print_button(WINDOW * win, const char *label, int y, int x, int selected);
                   ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:225:18: error: unknown type name 'WINDOW'
 void print_title(WINDOW *dialog, const char *title, int width);
                  ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:226:15: error: unknown type name 'WINDOW'
 void draw_box(WINDOW * win, int y, int x, int height, int width, chtype box,
               ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:226:66: error: unknown type name 'chtype'
 void draw_box(WINDOW * win, int y, int x, int height, int width, chtype box,
                                                                  ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:227:8: error: unknown type name 'chtype'
        chtype border);
        ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:228:18: error: unknown type name 'WINDOW'
 void draw_shadow(WINDOW * win, int y, int x, int height, int width);
                  ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c: In function 'conf':
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:97:29: error: 'KEY_MAX' undeclared (first use in this function)
 #define ERRDISPLAYTOOSMALL (KEY_MAX + 1)
                             ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c:671:45: note: in expansion of macro 'ERRDISPLAYTOOSMALL'
   if (res == 1 || res == KEY_ESC || res == -ERRDISPLAYTOOSMALL)
                                             ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:97:29: note: each undeclared identifier is reported only once for each function it appears in
 #define ERRDISPLAYTOOSMALL (KEY_MAX + 1)
                             ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c:671:45: note: in expansion of macro 'ERRDISPLAYTOOSMALL'
   if (res == 1 || res == KEY_ESC || res == -ERRDISPLAYTOOSMALL)
                                             ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c: In function 'show_help':
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c:793:27: error: 'stdscr' undeclared (first use in this function)
  help.max_width = getmaxx(stdscr) - 10;
                           ^
In file included from /opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c:23:0:
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c: In function 'conf_choice':
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/lxdialog/dialog.h:97:29: error: 'KEY_MAX' undeclared (first use in this function)
 #define ERRDISPLAYTOOSMALL (KEY_MAX + 1)
                             ^
/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf.c:855:9: note: in expansion of macro 'ERRDISPLAYTOOSMALL'
   case -ERRDISPLAYTOOSMALL:
         ^
Makefile:171: recipe for target 'mconf.o' failed
make[2]: *** [mconf.o] Error 1
make[2]: Leaving directory '/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig'
make[1]: *** No rule to make target '/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/conf-idf', needed by '/opt/nodemcu-firmware/sdk/esp32-esp-idf/tools/kconfig/mconf-idf'.  Stop.
make[1]: Leaving directory '/opt/nodemcu-firmware'
Makefile:10: recipe for target 'menuconfig' failed
make: *** [menuconfig] Error 2

> 
```

Can someone tell me what's wrong?

### PR DESCRIPTION
**Pull request by HHHartmann**
fixes #57

### CODE REVIEW

File: configure-esp32 [hunk 1]
```diff
@@ -4,4 +4,7 @@ set -e
 
 cd /opt/nodemcu-firmware
 
+# Basically revert windows lineends. Only needed if checkout is done under windows.
+git submodule foreach --recursive git checkout .
```
Comment (marcelstoer) [inline configure-esp32]: Can I suggest to wrap a condition around this? No need to burn extra CPU cycles on *nix platforms.

**Review by marcelstoer — CHANGES_REQUESTED** [review]

File: configure-esp32 (hunk 1 above)
Comment (HHHartmann) [reply configure-esp32]: Sure would like to but after a quick search I could not dig up anything that would tell me what host system I am on.
@marcelstoer do you have a hint here?

Comment (HHHartmann) [reply configure-esp32]: I could guess from the mountpoint of the firmware
Unter Windows it looks like ```//10.0.75.1/C on /opt/nodemcu-firmware type cifs```
what does it lock like on linux/Mac?

Comment (marcelstoer) [reply configure-esp32]: Clever idea! You could use `df|grep cifs` or `df|grep osxfs`.

However, I was initially thinking about detecting Windows newlines in a sample file (one we know must be there).

**Comment by marcelstoer** [comment]
Thanks again Gregor, I appreciate your contributions.

**Comment by HHHartmann** [comment]
I'm glad you like it!

**Pull request merged**